pydocstringformatter/formatting/base.py

```diff
@@ -5,6 +5,17 @@
 class Formatter:
     """Base class for docstring formatter."""
 
+    optional = False
+
+    @property
+    @abc.abstractmethod
+    def name(self) -> str:
+        """Name of the Formatter, choose wisely, this is an API.
+
+        This will be used to create argparse options when added to
+        'pydocstringformatter.formatting.FORMATTERS'.
```

Comment:
```suggestion
        """Name of the Formatter.

        This will be used to create argparse options when added to
        'pydocstringformatter.formatting.FORMATTERS'. Therefore, it is
        user-facing and should be chosen carefully.
```